enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -243,36 +243,33 @@ func (s *SociArtifactStore) GetArtifacts(ctx context.Context, req *socipb.GetArt
 		return nil, err
 	}
 
-	// Try to only read-pull-index-write once at a time to prevent hammering
-	// the containter registry with a ton of parallel pull requests, and save
-	// apps a bunch of parallel work.
-	workKey := fmt.Sprintf("soci-artifact-store-image-%s", configHash.Hex)
-	respBytes, err := s.deduper.Do(ctx, workKey, func() ([]byte, error) {
-		resp, err := s.getArtifactsFromCache(ctx, configHash)
-		if status.IsNotFoundError(err) {
+	resp, err := s.getArtifactsFromCache(ctx, configHash)
```

Comment:
remove the extra level of nesting

if err == nil {
  return resp, nil
}
if err != nil && !status.IsNotFoundError(err) {
  return nil, err
}

...the rest...

Reply:
nice, done